### PR TITLE
Tests of P16

### PR DESCRIPTION
**Pull request by kmizu**

### CODE REVIEW

File: src/test/scala/jp/co/dwango/s99/P16Check.scala [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package jp.co.dwango.s99
+
+import org.scalacheck.{Gen, Properties}
+import Gen.listOf, Gen.chooseNum
+import org.scalacheck.Prop.{BooleanOperators, forAll}
+
+class P16Check extends Properties("P16") {
+  property("drop()") = forAll(listOf(chooseNum(Int.MinValue, Int.MaxValue)), chooseNum(-1, 10)){ (s: List[Int], i: Int) =>
```
Comment (xuwei-k) [inline src/test/scala/jp/co/dwango/s99/P16Check.scala]: why `chooseNum(-1, 10)` ? should be `chooseNum(0, ` ?

File: src/test/scala/jp/co/dwango/s99/P16Check.scala [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    (0 < i && i <= s.length) ==> (P16.drop(i, s) == s.zipWithIndex.map{ case (e, j) =>(e, j +1) }.filterNot { case (e, j) =>
```
Comment (xuwei-k) [inline src/test/scala/jp/co/dwango/s99/P16Check.scala]: I think `0 < i` is unnecessary if `chooseNum(0, `

File: src/test/scala/jp/co/dwango/s99/P16Check.scala (hunk 1 above)
Comment (xuwei-k) [inline src/test/scala/jp/co/dwango/s99/P16Check.scala]: Oops, The problem 16 is 1 origin, not 0 origin !?
should be `chooseNum(1,` ?

File: src/test/scala/jp/co/dwango/s99/P16Check.scala (hunk 2 above)
Comment (pocketberserker) [inline src/test/scala/jp/co/dwango/s99/P16Check.scala]: I think `i <= s.length` is is unnecessary.

```scala
implicit val listGen = listOf(chooseNum(Int.MinValue, Int.MaxValue)
implicit val intGen = listGen.flatMap(l => chooseNum(0, l.length))
property("drop()") = forAll { ...
```

Comment (kmizu) [inline src/test/scala/jp/co/dwango/s99/P16Check.scala]: Thanks!  I'll fix them.

**Pull request merged**